rubyforgood/human-essentials
# 3429 reinviting accepted partners

# Checklist:
Checked, although I didn't add documentation

Resolves #3429

### Description
- This makes it so that if someone has accepted an invitation to the organization, a new one is not sent out when clicking on "Create User" in the partner coworker screen. 

 - The new behavior is being redirected to the coworker screen and a message appearing saying "(name) has already joined the organization

- Pros: does intended behavior
- Cons: added complexity and an additional check for the user before the invitation is sent

- no dependencies

### Type of change

* New feature (non-breaking change which adds functionality)
* This change requires a documentation update (I'm leaving this because I'm not sure it doesn't)

### How Has This Been Tested?

- Send an invitation to a new user
- In a separate browser, open the invitation link and fill out the forms to sign in to the new account (this Is what counts as "accepting" the invitation, and should maybe change?)
- In the original browser, invite the same person again (the email is what matters for identification)
- verify that the email isn't sent and a message says "(name) has already joined the organization"

### Screenshots
<img width="1119" alt="Screen Shot 2023-03-28 at 6 18 43 PM" src="https://user-images.githubusercontent.com/84088137/228379386-4f902537-cac2-4cfe-a771-d60977c7be66.png">
